### PR TITLE
samples: net: echo-client: Add overlay config support

### PR DESCRIPTION
**Pull request by jukkar**
Refactor the prj*conf files so that there is only one master
prj.conf and several overlay conf files that add / change
only minor subset of functionality.

Use the overlay files like so:
    cmake -DBOARD=nrf52840_pca10056 \
          -DOVERLAY_CONFIG=overlay-ot.conf ..

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: samples/net/echo_client/boards/nrf52840_pca10056.conf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+CONFIG_SYS_LOG_SHOW_COLOR=y
+CONFIG_PRINTK=y
+
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+CONFIG_FLASH=y
+CONFIG_FLASH_PAGE_LAYOUT=y
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
+
+CONFIG_NET_IPV4=n
```
Comment (rlubos) [inline samples/net/echo_client/boards/nrf52840_pca10056.conf]: Original config file for 802154 nRF does not enable TCP, therefore we should disable it here -> `CONFIG_NET_TCP=n`

**Review by rlubos — COMMENTED** [review]
Tested with raw 802154 and OpenThread on nRF52840. OpenThread worked just fine, raw 802154 needs to explicitly disable TCP in order to work.

**Review by dbkinder — APPROVED** [review]
doc +1

**Comment by jukkar** [comment]
Updated according to comments.

**Review by rlubos — APPROVED** [review]
Thanks!

**Pull request merged**